src/main/java/com/games/online/shop/service/dao/GameShopDao.java

```diff
@@ -46,4 +64,43 @@ public void deleteMechanics(Mechanics mechanics) {
             log.error("Error deleting mechanics mechanicsId={}", mechanics.getId(), ex);
         }
     }
+
+    public void addOrderItem(Long orderId, OrderItem orderItem) {
+        try {
+            Optional<Order> order = orderRepository.findById(orderId);
+            if (order.isPresent()) {
+                setOrderItemId(orderId, orderItem);
+                orderItem.setOrder(order.get());
+                order.get().addOrderItem(orderItem);
+                orderRepository.save(order.get());
+                orderItemRepository.save(orderItem);
+            }
```

Comment:
What should happen if the order isn't present? The method calling this item shouldn't think that it was successful, right?

Reply:
Good catch! Fixed 👍